enhancements/proxy/global-cluster-egress-proxy.md

```diff
@@ -0,0 +1,285 @@
+will confirm that the validation endpoints can be successfully reached using the new configuration
+before accepting the new configuration.  Once accepted, the configuration is moved into the status
+section of the proxy config resource.  Components should only consume the proxy configuration from
+this location.  
+
+Similarly, any user provided CAs will only be copied into an "accepted" CAs 
+configmap after confirming the validation endpoints can be accessed using the new CAs.
+
+Additional behaviors:
+
+* configmaps labeled with `config.openshift.io/inject-trusted-cabundle: "true"` will have the current 
+set of additional CAs injected into them by logic in the cluster network operator.
+* deployments with the `config.openshift.io/inject-proxy: <container-name>` will get the current proxy
+environment variables injected (HTTP_PROXY, HTTPS_PROXY, NO_PROXY) by the cluster version operator.
+
+Critical touch points for the administrator:
+
+* edits cluster scoped proxy config resource (spec fields)
+* provides a configmap of additional CAs
+
+Critical touch points for proxy configuration consumers:
+
+* Operator consumes status fields from cluster scoped proxy config resource and updates its operand accordingly
+* *May* consume the "accepted CAs" configmap (openshift-config-managed/trusted-ca-bundle) to get CAs
+* *Should* create their own configmap with the `config.openshift.io/inject-trusted-cabundle: "true"` label and
+consume the CA bundle from there.
+* Operator deployment may request proxy environment injection via the `config.openshift.io/inject-proxy: <container-name>`
+annotation since operators cannot control their own environment variables, but the oprator is responsible for mounting
+a configmap to pick up the CAs if it needs them.
+
+Automatic no proxy behavior:
+
+The following domains/hosts are automatically added to the no proxy configuration:
+* TODO
+
+Known limitations/future enhancements:
+
+* The no proxy entries that are automatically added do not include the cluster domain generically, so
+requests that go to external routes will likely go through the proxy.  This may be undesirable, though
+it should not break as long as the proxy is able to call back to the cluster.  The administrator can add 
+additional no proxy entries.  It was deemed unacceptable to automatically noproxy the entire cluster domain 
+because the front end load balancing/routing for the cluster could reside outside the clusters network.
+* Not all cloud service apis are added to the no proxy list.  This means some cloud api requests may
+go through the proxy, which again is undesirable though it should not break things as long as the proxy
+is functional.  The administrator can add additional no proxy entries.
+* No validation endpoints can be provided at install time.  This is because there is no point in validating
+the proxy configuration during install.  If it is incorrect, the install will fail.  However it also 
```

Comment:
Installs take a long time. By providing validation endpoint support during install, the bad config/proxy can be detected earlier.

Reply:
will note.